### PR TITLE
Azure_IaC, new rego files (part2)

### PR DESCRIPTION
**Pull request by shahinshirvani**

### CODE REVIEW

File: azure/iac/master-compliance-test.json [hunk 1]
```diff
@@ -2524,7 +2524,7 @@
                             "id": "PR-AZR-0125-ARM",
                             "eval": "data.rule.protocol",
                             "message": "data.rule.protocol_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure ApplicationGateway has (properties.httpListeners.properties.protocol='Https')",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2527]: **See my validation message at line 2432 (for id PR-AZR-0107-ARM)**. make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 2]
```diff
@@ -2555,7 +2555,7 @@
                             "id": "PR-AZR-0122-ARM",
                             "eval": "data.rule.region",
                             "message": "data.rule.region_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure StorageAccount has (location='northeurope' or location='westeurope')",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2558]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 3]
```diff
@@ -2586,7 +2586,7 @@
                             "id": "PR-AZR-PR-AZR-0112-ARM-ARM",
                             "eval": "data.rule.blobService",
                             "message": "data.rule.blobService_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure StorageAccount has ( properties.encryption.services with [ name='blob' and enabled=true ] )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2589]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 4]
```diff
@@ -2621,7 +2621,7 @@
                             "id": "PR-AZR-0113-ARM",
                             "eval": "data.rule.fileService",
                             "message": "data.rule.fileService_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure StorageAccount has ( properties.encryption.services with [ name='file' and enabled=true ] )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2624]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 5]
```diff
@@ -2656,7 +2656,7 @@
                             "id": "PR-AZR-0114-ARM",
                             "eval": "data.rule.keySource",
                             "message": "data.rule.keySource_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure StorageAccount has (properties.encryption.keySource='Microsoft.Keyvault')",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2659]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 6]
```diff
@@ -2674,6 +2674,130 @@
...
+                        "ARM_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(sql_servers_encryption.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-0111-ARM",
+                            "eval": "data.rule.serverKeyType",
+                            "message": "data.rule.serverKeyType_err",
+                            "remediationDescription": "Make sure SQLServer/encryptionProtector has ( properties.serverKeyType ='AzureKeyVault' )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2690]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 7]
```diff
@@ -2674,6 +2674,130 @@
...
+                        "ARM_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(postgreSQL.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-0115-ARM",
+                            "eval": "data.rule.geoRedundantBackup",
+                            "message": "data.rule.geoRedundantBackup_err",
+                            "remediationDescription": "Make sure PostgreSQL has ( properties.storageProfile.geoRedundantBackup = 'Enabled' )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2721]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 8]
```diff
@@ -2674,6 +2674,130 @@
...
+                        "ARM_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(postgreSQL.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-0124-ARM",
+                            "eval": "data.rule.sslEnforcement",
+                            "message": "data.rule.sslEnforcement_err",
+                            "remediationDescription": "Make sure PostgreSQL has ( properties.sslEnforcement = 'Enabled' )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2752]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM)**. make the description format like this one.

File: azure/iac/master-compliance-test.json [hunk 9]
```diff
@@ -2674,6 +2674,130 @@
...
+                        "ARM_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(dbAccounts.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-0105-ARM",
+                            "eval": "data.rule.tagsLength",
+                            "message": "data.rule.tagsLength_err",
+                            "remediationDescription": "Make sure CosmosDbAccount has ( tags length() > 0 )",
```
Comment (rezoan) [inline azure/iac/master-compliance-test.json:2783]: **See my validation message at line 2496 (for id PR-AZR-0109-ARM).** make the description format like this one.

**Review by rezoan — CHANGES_REQUESTED** [review]
Please see my review messages on each file and change accordingly.

**Review by farchide — APPROVED** [review]

File: azure/iac/dbAccounts.rego [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+azure_issue ["tagsLength"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.documentdb/databaseaccounts"
+    count(resource.tags) == 0
+}
+
+tagsLength {
+    lower(input.resources[_].type) == "microsoft.documentdb/databaseaccounts"
+    not azure_issue["tagsLength"]
+    not azure_attribute_absence["tagsLength"]
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/dbAccounts.rego:23]: fixed

File: azure/iac/dbAccounts.rego [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    azure_issue["tagsLength"]
+}
+
+tagsLength = false {
+    azure_attribute_absence["tagsLength"]
+}
+
+
+tagsLength_err = "Ensure that Cosmos DB Account has an associated tag" {
+    azure_issue["tagsLength"]
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/dbAccounts.rego:37]: fixed

File: azure/iac/dbAccounts.rego [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+tagsLength_miss_err = "Ensure that Cosmos DB Account has an associated tag" {
+    azure_attribute_absence["tagsLength"]
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/dbAccounts.rego:41]: fixed

File: azure/iac/postgreSQL.rego [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+azure_issue ["sslEnforcement"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.dbforpostgresql/servers"
+    lower(resource.properties.sslEnforcement) != "enabled"
+}
+
+sslEnforcement {
+    lower(input.resources[_].type) == "microsoft.dbforpostgresql/servers"
+    not azure_issue["sslEnforcement"]
+    not azure_attribute_absence["sslEnforcement"]
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego:76]: fixed

File: azure/iac/master-compliance-test.json [hunk 14]
```diff
@@ -2429,7 +2429,7 @@
                             "id": "PR-AZR-0107-ARM",
                             "eval": "data.rule.KeyVault",
                             "message": "data.rule.KeyVault_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure keyVault has (properties.accessPolicies.permissions.keys or properties.accessPolicies.permissions.secrets or properties.accessPolicies.permissions.certificates)",
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/master-compliance-test.json]: fixed

File: azure/iac/master-compliance-test.json (hunk 5 above)
Comment (shahinshirvani) [reply azure/iac/master-compliance-test.json]: fixed

File: azure/iac/master-compliance-test.json [hunk 15]
```diff
@@ -2462,7 +2462,7 @@
                             "id": "PR-AZR-0108-ARM",
                             "eval": "data.rule.enableSoftDelete",
                             "message": "data.rule.enableSoftDelete_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure keyVault has (properties.enableSoftDelete=true)",
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/master-compliance-test.json]: fixed

File: azure/iac/master-compliance-test.json [hunk 16]
```diff
@@ -2493,7 +2493,7 @@
                             "id": "PR-AZR-0109-ARM",
                             "eval": "data.rule.enablePurgeProtection",
                             "message": "data.rule.enablePurgeProtection_err",
-                            "remediationDescription": "",
+                            "remediationDescription": "Make sure keyVault has (properties.enablePurgeProtection=true)",
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/master-compliance-test.json]: fixed

File: azure/iac/postgreSQL.rego [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+sslEnforcement_miss_err = "Ensure 'Enforce SSL connection' is set to 'ENABLED' for PostgreSQL Database Server" {
+    azure_attribute_absence["sslEnforcement"]
+}
+
+sslEnforcement_metadata := {
+    "Policy Code": "PR-AZR-0124-ARM",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Ensure 'Enforce SSL connection' is set to 'ENABLED' for PostgreSQL Database Server",
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/sql_servers_encryption.rego [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+serverKeyType {
+    lower(input.resources[_].type) == "microsoft.sql/servers/encryptionprotector"
+    not azure_issue["serverKeyType"]
+}
+
+serverKeyType = false {
+    azure_issue["serverKeyType"]
+}
+
+serverKeyType_err = "Ensure SQL server's TDE protector is encrypted with Customer-managed key" {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/sql_servers_encryption.rego]: fixed

File: azure/iac/postgreSQL.rego [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+sslEnforcement = false {
+    azure_issue["sslEnforcement"]
+}
+
+sslEnforcement = false {
+    azure_attribute_absence["sslEnforcement"]
+}
+
+
+sslEnforcement_err = "Ensure 'Enforce SSL connection' is set to 'ENABLED' for PostgreSQL Database Server" {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/postgreSQL.rego [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+geoRedundantBackup_miss_err = "Ensure that Geo Redundant Backups is enabled on PostgreSQL" {
+    azure_attribute_absence["geoRedundantBackup"]
+}
+
+geoRedundantBackup_metadata := {
+    "Policy Code": "PR-AZR-0115-ARM",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Ensure that Geo Redundant Backups is enabled on PostgreSQL",
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/postgreSQL.rego [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+sslEnforcement = false {
+    azure_attribute_absence["sslEnforcement"]
+}
+
+
+sslEnforcement_err = "Ensure 'Enforce SSL connection' is set to 'ENABLED' for PostgreSQL Database Server" {
+    azure_issue["sslEnforcement"]
+}
+
+sslEnforcement_miss_err = "Ensure 'Enforce SSL connection' is set to 'ENABLED' for PostgreSQL Database Server" {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/postgreSQL.rego [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+
+geoRedundantBackup = false {
+    azure_attribute_absence["geoRedundantBackup"]
+}
+
+geoRedundantBackup = false {
+    azure_issue["geoRedundantBackup"]
+}
+
+geoRedundantBackup_err = "Ensure that Geo Redundant Backups is enabled on PostgreSQL" {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/postgreSQL.rego [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+    azure_issue["geoRedundantBackup"]
+}
+
+geoRedundantBackup_miss_err = "Ensure that Geo Redundant Backups is enabled on PostgreSQL" {
```
[the remark this answers is not in the document]
Comment (shahinshirvani) [reply azure/iac/postgreSQL.rego]: fixed

File: azure/iac/master-compliance-test.json (hunk 1 above)
Comment (shahinshirvani) [reply azure/iac/master-compliance-test.json]: fixed

**Review by farchide — APPROVED** [review]

**Review by rezoan — APPROVED** [review]
Looks good. Approved.

**Pull request merged**